internal/backend_helpers/gcloud.go

```diff
@@ -421,62 +479,10 @@ func (helper *GCloudHelper) RunPeriodic(ctx context.Context, env map[string]inte
 					helper.informHSM[xname] = 0
 				}
 
-				// Note the discovery for later
+				// Note the discovery for later to prevent removal from known instances
 				seen[xname] = true
-
 				// Process the instance...
```

Comment:
REVIEWERS: this logic got moved to updateKnownInstance() above so that it gets run both on periodic discovery and on queried discovery.